src/Pauser.sol

```diff
@@ -0,0 +1,320 @@
+     */
+    mapping(bytes32 id => uint256) public unpauseExecutionTime;
+
+    /**
+     * @notice Timestamp indicating an operation has been executed.
+     */
+    uint256 public constant _DONE_TIMESTAMP = uint256(1);
+
+    /**
+     * @notice Global Role registry contract.
+     */
+    RoleRegistry public roleRegistry;
+
+    //============================== ROLES ================================
+
+    /**
+     * @notice Contract specific roles.
+     */
+    bytes32 public constant PAUSER_ADMIN = keccak256("PAUSER_ADMIN");
+    /**
+     * @notice Protocol specific roles.
+     */
+    bytes32 public constant PROTOCOL_PAUSER = keccak256("PROTOCOL_PAUSER");
+    bytes32 public constant PROTOCOL_UNPAUSER = keccak256("PROTOCOL_UNPAUSER");
+    bytes32 public constant PROTOCOL_UPGRADER = keccak256("PROTOCOL_UPGRADER");
+
+    //============================== ERRORS ===============================
+
+    error Pauser__IndexOutOfBounds();
+
+    //============================== EVENTS ===============================
+
+    event PausablePaused(address indexed pausable);
+    event PausableUnpaused(address indexed pausable);
+    event PausableAdded(address indexed pausable);
+    event PausableRemoved(address indexed pausable);
+
+    event MinUnpauseDelayUpdated(uint256 oldDuration, uint256 newDuration);
+    event UnpauseScheduled(bytes32 indexed id, uint256 indexed timestampProposed, uint256 timestampExecutable);
+    event UnpauseExecuted(bytes32 indexed id);
+    event UnpauseDeleted(bytes32 indexed id);
+    
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    // ========================================= ADMIN FUNCTIONS =========================================
+
+    function initialize(IPausable[] memory _pausables, uint256 _minUnpauseDelay, address _roleRegistry) public initializer {
+        for (uint256 i = 0; i < _pausables.length; ++i) {
+            pausables.push(_pausables[i]);
+        }
+
+        minUnpauseDelay = _minUnpauseDelay;
+        emit MinUnpauseDelayUpdated(0, _minUnpauseDelay);
+
+        roleRegistry = RoleRegistry(_roleRegistry);
+
+        __Ownable_init();
+        __UUPSUpgradeable_init();
+    }
+
+    /**
+     * @notice Adds a contract to the list of pausables.
+     */
+    function addPausable(IPausable _pausable) external onlyRole(PAUSER_ADMIN) {
+        pausables.push(_pausable);
+        emit PausableAdded(address(_pausable));
+    }
+
+    /**
+     * @notice Removes a contract from the list of pausables.
+     */
+    function removePausable(uint256 _index) external onlyRole(PAUSER_ADMIN) {
+        uint256 pausablesLength = pausables.length;
+        if (_index >= pausablesLength) {
+            revert Pauser__IndexOutOfBounds();
+        }
+        address removed = address(pausables[_index]);
+        pausables[_index] = pausables[pausablesLength - 1];
```

Comment:
while a swap remove is efficient, I think it breaks the semantics you have established elsewhere in the contract. All your events are based around an indexed `index` field. After performing this operation, anyone relying on those events would no longer have a valid view of the world

Reply:
Lets add a convenience function that helps you find this value. 
`indexOf(address)` or something to that effect. Not the most effecient onchain but would be handy offchain and via etherscan when you want to perform a remove operation

Reply:
added ✅